### ISSUE TITLE
[bug]type checker report wrong infer type

### ISSUE THREAD

**Issue by SolaWing** [castwide/solargraph#264]
```ruby
# frozen_string_literal: true

class A
  def initialize
    @a = 1
  end

  # @return [Boolean]
  def sss? # ERROR: report @return type `Boolean` does not match inferred type `Integer`
    b = @a == 1 # NOTE: b infer to Boolean
    @a == 1
  end
end
```

by the way, can we have a feature, only report the mismatch type, but ignore the uptyped warning? so I don't have to comment every simple code, just add what I need. sorbet `typed: true` is this behaviour

**Comment by castwide** [comment]
The master branch has some updates that fix a bug in method type inference. I expect to release them in gem 0.38.3 tomorrow.

It makes sense to ignore missing `@return` tags when Solargraph can infer a return type on its own. I'll look into making it an option.

**Comment by castwide** [comment]
I posted a proposal for type checking levels in #192. The "normal" level would use your proposal of allowing untyped methods but validating the `@return` type if it exists.

**Comment by castwide** [comment]
Released in 0.38.3.
